pkg/forwarder/forwarder.go

```diff
@@ -354,12 +317,16 @@ func (f *DefaultForwarder) createPriorityHTTPTransactions(endpoint endpoint, pay
 				t.Domain = domain
 				t.Endpoint = transactionEndpoint
 				t.Payload = payload
+				t.endpointName = endpoint.name
```

Comment:
since you're storing all the fields of the `endpoint` in `Transaction`, I think you should simply store the `endpoint` in `t.Endpoint` (and update the type of that field accordingly)

Reply:
The `endpoint` struct misses the domain, though.